### PR TITLE
easier_way_provide_encoding

### PR DESCRIPTION
**Pull request by fischey**
provide an easier using an encoding while writing a java file to a directory.

### CODE REVIEW

**Comment by fischey** [comment]
Might be a fix for this issue: https://github.com/square/javapoet/issues/564.

File: src/main/java/com/squareup/javapoet/JavaFile.java [hunk 1]
```diff
@@ -83,6 +84,11 @@ public void writeTo(Appendable out) throws IOException {
 
   /** Writes this to {@code directory} as UTF-8 using the standard directory structure. */
   public void writeTo(Path directory) throws IOException {
+    writeTo(directory, UTF_8);
+  }
+
+  /** Writes this to {@code directory} as UTF-8 using the standard directory structure. */
```
Comment (tbroyer) [inline src/main/java/com/squareup/javapoet/JavaFile.java]: Javadoc needs fixing as this is no longer utf8 only.

File: src/main/java/com/squareup/javapoet/JavaFile.java [hunk 2]
```diff
@@ -83,6 +84,11 @@ public void writeTo(Appendable out) throws IOException {
...
+  public void writeTo(Path directory, final Charset charset) throws IOException {
```
Comment (JakeWharton) [inline src/main/java/com/squareup/javapoet/JavaFile.java]:
```suggestion
  public void writeTo(Path directory, Charset charset) throws IOException {
```

File: src/main/java/com/squareup/javapoet/JavaFile.java (hunk 1 above)
Comment (fischey) [reply src/main/java/com/squareup/javapoet/JavaFile.java]: Thank you for the hint! :-)

File: src/main/java/com/squareup/javapoet/JavaFile.java (hunk 2 above)
Comment (fischey) [reply src/main/java/com/squareup/javapoet/JavaFile.java]: Thank you for the hint. Its done.

File: src/main/java/com/squareup/javapoet/JavaFile.java [hunk 3]
```diff
@@ -83,6 +84,11 @@ public void writeTo(Appendable out) throws IOException {
 
   /** Writes this to {@code directory} as UTF-8 using the standard directory structure. */
   public void writeTo(Path directory) throws IOException {
+    writeTo(directory, UTF_8);
+  }
+
+  /** Writes this to {@code directory} with the provided {@code charset} using the standard directory structure. */
```
Comment (Egorand) [inline src/main/java/com/squareup/javapoet/JavaFile.java]: Looks like checkstyle is complaining that this line is longer than the limit, which is 100 lines. Can you please fix?

Comment (fischey) [reply src/main/java/com/squareup/javapoet/JavaFile.java]: I corrected this. Sorry that I not see this in the first place.

Comment (Egorand) [reply src/main/java/com/squareup/javapoet/JavaFile.java]: Thanks for fixing!

**Comment by fischey** [comment]
Thank you for accepting the change :-)
I signed your CLA. So you can proceed. Best regards.

**Pull request merged**